agent/engine/docker_task_engine.go

```diff
@@ -303,6 +308,14 @@ func (engine *DockerTaskEngine) sweepTask(task *api.Task) {
 			seelog.Errorf("Error removing container reference from image state: %v", err)
 		}
 	}
+
+	// Clean metadata directory for task
+	if engine.cfg.ContainerMetadataEnabled {
```

Comment:
I am not sure why an additional check is needed. If the file is not created, the cleanup would simply do nothing (Not even return an error)
https://golang.org/pkg/os/#RemoveAll